contracts/dappnodeContracts/IncentiveDepositContract.sol

```diff
@@ -114,11 +116,10 @@ contract IncentiveDepositContract is OwnableUpgradeable {
 
         addressToIncentive[msg.sender].isClaimed = true;
 
-        sbcToken.transferAndCall(
-            sbcDepositContract,
-            DEPOSIT_AMOUNT * validatorNum,
-            data
-        );
+        // Note: requires this contract to have pre-approved GNO transfers to sbcDepositContract
+        // Since the sbcDepositContract is trusted, it should assume an infinite approve
+        (bool success, bytes memory _data) = sbcDepositContract.call(data);
```

Comment:
This could be extremely dangerous! couldn't a user try to drain all the funds tricking this call?
We should always call to a known interface and hardcode the `stake_amount`
Maybe we should update the callData of this function aswell to match the interface:
```
   bytes memory pubkey,
        bytes memory withdrawal_credentials,
        bytes memory signature,
        bytes32 deposit_data_root,
        uint256 stake_amount // hardcoded

Reply:
Yeah agree the architecture is not hardened enough. I put the PR as draft as the entire approach of the program is flawed now that withdrawals will activate. Let's revisit latter